x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/add_source/github_via_app_logic.ts

```diff
@@ -0,0 +1,115 @@
+import { HttpLogic } from '../../../../../shared/http';
+import { KibanaLogic } from '../../../../../shared/kibana';
+import { AppLogic } from '../../../../app_logic';
+import {
+  GITHUB_VIA_APP_SERVICE_TYPE,
+  GITHUB_ENTERPRISE_SERVER_VIA_APP_SERVICE_TYPE,
+} from '../../../../constants';
+import { SOURCES_PATH, getSourcesPath } from '../../../../routes';
+import { ContentSourceFullData } from '../../../../types';
+
+interface GithubViaAppValues {
+  githubAppId: string;
+  githubEnterpriseServerUrl: string;
+  stagedPrivateKey: string | null;
+  isSubmitButtonLoading: boolean;
+  indexPermissionsValue: boolean;
+}
+
+interface GithubViaAppActions {
+  setGithubAppId(githubAppId: string): string;
+  setGithubEnterpriseServerUrl(githubEnterpriseServerUrl: string): string;
+  setStagedPrivateKey(stagedPrivateKey: string | null): string | null;
+  setButtonNotLoading(): void;
+  createContentSource(): void;
+  setSourceIndexPermissionsValue(indexPermissionsValue: boolean): boolean;
+}
+
+export const GithubViaAppLogic = kea<MakeLogicType<GithubViaAppValues, GithubViaAppActions>>({
+  path: ['enterprise_search', 'workplace_search', 'github_via_app_logic'],
+  actions: {
+    setGithubAppId: (githubAppId: string) => githubAppId,
+    setGithubEnterpriseServerUrl: (githubEnterpriseServerUrl: string) => githubEnterpriseServerUrl,
+    createContentSource: () => true,
+    setStagedPrivateKey: (stagedPrivateKey: string) => stagedPrivateKey,
+    setButtonNotLoading: () => false,
+    setSourceIndexPermissionsValue: (indexPermissionsValue: boolean) => indexPermissionsValue,
+  },
+  reducers: {
+    githubAppId: [
+      '',
+      {
+        setGithubAppId: (_, githubAppId) => githubAppId,
+      },
+    ],
+    githubEnterpriseServerUrl: [
+      '',
+      {
+        setGithubEnterpriseServerUrl: (_, githubEnterpriseServerUrl) => githubEnterpriseServerUrl,
+      },
+    ],
+    stagedPrivateKey: [
+      null,
+      {
+        setStagedPrivateKey: (_, stagedPrivateKey) => stagedPrivateKey,
+      },
+    ],
+    isSubmitButtonLoading: [
+      false,
+      {
+        createContentSource: () => true,
+        setButtonNotLoading: () => false,
+      },
+    ],
+    indexPermissionsValue: [
+      true,
+      {
+        setSourceIndexPermissionsValue: (_, indexPermissionsValue) => indexPermissionsValue,
+        resetSourceState: () => false,
+      },
+    ],
+  },
+  listeners: ({ actions, values }) => ({
+    createContentSource: async () => {
+      const { isOrganization } = AppLogic.values;
+      const route = isOrganization
+        ? '/internal/workplace_search/org/create_source'
+        : '/internal/workplace_search/account/create_source';
+
+      const { githubAppId, githubEnterpriseServerUrl, stagedPrivateKey } = values;
+
+      const params = {
+        service_type: githubEnterpriseServerUrl
+          ? GITHUB_ENTERPRISE_SERVER_VIA_APP_SERVICE_TYPE
+          : GITHUB_VIA_APP_SERVICE_TYPE,
```

Comment:
Chenhui discovered a bug: if Base URL field is empty, we always send GITHUB_ENTERPRISE_SERVER_VIA_APP_SERVICE_TYPE to the backend. I'll fix it today

Reply:
Addressed here: https://github.com/elastic/kibana/pull/119764/commits/f4d1b71b6cc495ebc379443ab7dc8d6a58f66898